### PR TITLE
Update components' tables of arguments

### PR DESCRIPTION
**Pull request by igloosi**
Ensuring table of arguments for each components is accurate.

Updates to  `index.njk` and `README.md` files for each component.

Trello ticket: https://trello.com/c/rRQl78KY/1038-make-sure-all-component-argument-tables-are-accurate

For reviewers: Might be easiest to go through each component page on Github in the [branch](https://github.com/alphagov/govuk-frontend/tree/update-tables-of-arguments) to check the rendered README file.

### CODE REVIEW

File: src/components/footer/index.njk [hunk 1]
```diff
@@ -75,12 +103,12 @@
...
       },
       {
-        text: 'Array of items for use in the navigation section of the footer'
+        text: 'Array of items for use in the navigation section of the footer.'
       }
     ],
     [
       {
-        text: 'navigation[].title'
+        text: 'navigation.title'
```
Comment (nickcolley) [inline src/components/footer/index.njk]: Should this be consistent with the with the other approach that uses `{}.prop`?

Comment (igloosi) [reply src/components/footer/index.njk]: updated, but it's tricky as its [].{}.prop - don't want to complicate it too much

File: src/components/footer/index.njk [hunk 2]
```diff
@@ -117,7 +145,35 @@
...
+      {
+        text: 'No'
+      },
+      {
+        text: 'List item text in the navigation section of the footer.'
+      }
+    ],
+    [
+      {
+        text: 'navigation.items.[].href'
```
Comment (nickcolley) [inline src/components/footer/index.njk]: I think this is still inconsistent with other places using 'item.{}.prop'?

Comment (igloosi) [reply src/components/footer/index.njk]: yep, fixed it now

File: src/components/checkboxes/index.njk [hunk 3]
```diff
@@ -202,7 +230,21 @@
         text: 'No'
       },
       {
-        text: 'If true, checkbox will be disabled'
+        text: 'If true, checkbox will be disabled.'
+      }
```
Comment (hannalaakso) [inline src/components/checkboxes/index.njk]: I think we might need `items.{}.value` as well.

File: src/components/date-input/index.njk [hunk 4]
```diff
@@ -133,6 +133,34 @@
       {
         text: 'Arguments for the fieldset component (e.g. legend). See fieldset component.'
       }
+    ],
+    [
```
Comment (hannalaakso) [inline src/components/date-input/index.njk]: I think we might need `hint` object as well.

File: src/components/fieldset/index.njk [hunk 5]
```diff
@@ -67,7 +67,7 @@
     ],
     [
       {
-        text: 'legend.text'
+        text: 'legend.text (or) legend.html'
```
Comment (hannalaakso) [inline src/components/fieldset/index.njk]: These should probably be `legend.{}.text` etc. to be consistent.

File: src/components/footer/index.njk [hunk 6]
```diff
@@ -117,7 +145,35 @@
...
+        text: 'navigation.items.{}.href'
+      },
+      {
+        text: 'string'
+      },
+      {
+        text: 'No'
+      },
+      {
+        text: 'List item href atrribute in the navigation section of the footer. Both `text` and `href` attributes need to be present to create a link.'
```
Comment (hannalaakso) [inline src/components/footer/index.njk]: Nit: typo in "attribute".

File: src/components/radios/index.njk [hunk 7]
```diff
@@ -204,6 +232,20 @@ Let users select a single option from a list.
         text: 'If true, radio will be disabled.'
       }
     ],
+    [
```
Comment (hannalaakso) [inline src/components/radios/index.njk]: I think we need `items.{}.value` as well.

File: src/components/select/index.njk [hunk 8]
```diff
@@ -150,7 +136,7 @@ The HTML <code>&lt;select&gt;</code> element represents a control that provides
     ],
     [
       {
-        text: 'label'
+        text: 'item.label'
```
Comment (hannalaakso) [inline src/components/select/index.njk]: Hmmm should this be just `label`? It doesn't look like `items` uses `label` in the template.

File: src/components/table/index.njk [hunk 9]
```diff
@@ -104,40 +118,54 @@
...
-        text: 'string'
+        text: 'number'
+      },
+      {
+        text: 'No'
+      },
+      {
+        text: 'Specify how many columns a cell extends.'
+      }
+    ],
```
Comment (hannalaakso) [inline src/components/table/index.njk]: I think this might also need `head.[].format`.

File: src/components/footer/index.njk (hunk 6 above)
Comment (igloosi) [reply src/components/footer/index.njk]: can't see the typo

Comment (hannalaakso) [reply src/components/footer/index.njk]: It's in "atrribute" - sorry should have said that.

File: src/components/radios/index.njk (hunk 7 above)
Comment (igloosi) [reply src/components/radios/index.njk]: added

File: src/components/date-input/index.njk (hunk 4 above)
Comment (igloosi) [reply src/components/date-input/index.njk]: added

**Comment by igloosi** [comment]
@hannalaakso updated with comments

**Pull request merged**